src/diffusers/models/attention_processor.py

```diff
@@ -2185,9 +2189,20 @@ def __call__(
         hidden_states = torch.bmm(attention_probs, value)
         hidden_states = attn.batch_to_head_dim(hidden_states)
 
+        if ip_adapter_masks is not None:
+            if not isinstance(ip_adapter_masks, torch.Tensor) or ip_adapter_masks.ndim != 4:
+                raise ValueError(" ip_adapter_mask should be a tensor with shape [num_ip_adapter, 1, height, width]."
+                                 " Please use `IPAdapterMaskProcessor` to preprocess your mask") 
+            if len(ip_adapter_masks) != len(ip_hidden_states):
+                raise ValueError(
+                    f"Number of ip_adapter_masks ({len(ip_adapter_masks)}) must match number of IP-Adapters ({len(self.scale)})"
+                )
```

Comment:
I find it a little weird that in the condition we have the length of `ip_hidden_states` while in the error message, we're relying on the length of `self.scale`. Maybe, settle on one of the two here?